### PR TITLE
T-11: Create contacts page

### PR DESCRIPTION
**Pull request by VladyslavGilyuk**

### CODE REVIEW

File: src/components/common/table/customToolbar/index.tsx [hunk 1]
```diff
@@ -85,38 +97,41 @@ const CustomToolbar: React.FC<IProps> = ({ apiUrl, selectedPriorities, setSelect
...
-                    checked={selectedPriorities.includes(option)}
-                    onChange={() => handlePriorityFilter(option)}
-                  />
-                }
-                label={option}
-              />
-            ))}
-          </FilterSelect>
-        </FormControl>
+        {!disabledFilter ? (
```
Comment (mazhula97) [inline src/components/common/table/customToolbar/index.tsx]: && instead of ? :

File: src/components/common/table/index.tsx [hunk 2]
```diff
@@ -1,53 +1,81 @@
...
+  onDelete,
+  columns,
+  sortingOptions,
+  priorityOptions,
+  disabledFilter,
+  contactsForm,
+}) => {
   const dispatch = useAppDispatch();
 
+  const storeItems = useSelector(items);
```
Comment (mazhula97) [inline src/components/common/table/index.tsx]: probably it's better to put totalAmount and items in one object in the createSelector function?

File: src/components/contacts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import Table from '../common/table';
+import { columns, sortingOptions } from './helper';
+import { contacts, total } from '../../store/contacts/selector';
+import { deleteContactAsync, fetchContactAsync } from '../../store/contacts/thunk';
+
+const ContactsTable = () => {
+  return (
+    <Table
+      columns={columns}
+      fetch={fetchContactAsync}
```
Comment (mazhula97) [inline src/components/contacts/index.tsx]: `fetch` -> `fetchAction`

File: src/components/contacts/index.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      items={contacts}
+      total={total}
+      onDelete={deleteContactAsync}
```
Comment (mazhula97) [inline src/components/contacts/index.tsx]: `deleteAction`

File: src/components/contacts/index.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      sortingOptions={sortingOptions}
+      disabledFilter
+      contactsForm
```
Comment (mazhula97) [inline src/components/contacts/index.tsx]: lets replace it with formType which will use Enum like:
enum EFromType {
Contacts = "Contacts",
Tickets = "Tickets"
}
Then you can use this types everywhere you want

**Review by mazhula97 — CHANGES_REQUESTED** [review]

File: src/components/common/table/customToolbar/index.tsx (hunk 1 above)
Comment (VladyslavGilyuk) [reply src/components/common/table/customToolbar/index.tsx]: fix:  '&&' used instead of '?'

File: src/components/contacts/index.tsx (hunk 3 above)
Comment (VladyslavGilyuk) [reply src/components/contacts/index.tsx]: fix: renamed to fetchAction in Contacts, Tickets and Table folders

File: src/components/contacts/index.tsx (hunk 4 above)
Comment (VladyslavGilyuk) [reply src/components/contacts/index.tsx]: fix: renamed to deleteAction in Contacts, Tickets and Table folders

File: src/components/contacts/index.tsx (hunk 5 above)
Comment (VladyslavGilyuk) [reply src/components/contacts/index.tsx]: fix: remade with Enum

File: src/components/common/table/index.tsx (hunk 2 above)
Comment (VladyslavGilyuk) [reply src/components/common/table/index.tsx]: fix: done with createSelector and imported as one object

File: src/store/contacts/selector.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { RootState } from '..';
+import { createSelector } from '@reduxjs/toolkit';
+
+export const contacts = (state: RootState) => state.contacts.contacts;
+export const total = (state: RootState) => state.contacts.total;
+export const contactsStore = createSelector([contacts, total], (storeItems, storeTotal) => ({
```
Comment (mazhula97) [inline src/store/contacts/selector.ts]: Let's call it more clear contacts = {data, total or contacts, total}

File: src/store/tickets/selector.ts [hunk 7]
```diff
@@ -1,4 +1,9 @@
 import { RootState } from '..';
+import { createSelector } from '@reduxjs/toolkit';
 
 export const tickets = (state: RootState) => state.tickets.tickets;
 export const total = (state: RootState) => state.tickets.total;
+export const ticketsStore = createSelector([tickets, total], (storeItems, storeTotal) => ({
```
Comment (mazhula97) [inline src/store/tickets/selector.ts]: the same as above

File: src/components/common/table/index.tsx [hunk 8]
```diff
@@ -1,53 +1,87 @@
...
-  const storeTickets = useSelector(tickets);
-  const storeTotal = useSelector(total);
+  const [selectedItem, setSelectedItem] = useState<ITicketState | IContactState | null>(null);
 
   const handleUpdateItem = useCallback(
     (id?: string) => {
-      const ticketToEdit = storeTickets?.find((ticket) => ticket.id === id) || null;
-      setSelectedTicket(ticketToEdit);
+      const itemToEdit =
+        (storeItems as (ITicketState | IContactState)[])?.find((item) => item.id === id) || null;
```
Comment (mazhula97) [inline src/components/common/table/index.tsx]: I think that you can use if(storeItems && storeItems.length)

File: src/components/common/table/index.tsx
Comment (mazhula97) [inline src/components/common/table/index.tsx:1]: Let's create a new one custom hook `useTable` where we'll use `useSortFilter` hook and store all from the component

**Review by mazhula97 — CHANGES_REQUESTED** [review]

File: src/store/contacts/selector.ts (hunk 6 above)
Comment (VladyslavGilyuk) [reply src/store/contacts/selector.ts]: fix: renamed to data, total

File: src/store/tickets/selector.ts (hunk 7 above)
Comment (VladyslavGilyuk) [reply src/store/tickets/selector.ts]: fix: renamed to data, total

File: src/components/common/table/index.tsx (hunk 8 above)
Comment (VladyslavGilyuk) [reply src/components/common/table/index.tsx]: No, it doesnt work.

File: src/components/common/table/index.tsx
Comment (VladyslavGilyuk) [reply src/components/common/table/index.tsx:1]: feat: useTable hook created

**Review by mazhula97 — APPROVED** [review]

**Pull request merged**